src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const groupKey = item[key] as string;
```

Comment:
There is an assumption here that `item[key]` will always be a string, which might not be true for every T. This could lead to a runtime error if `item[key]` is not a string. To ensure type safety, consider adding a type guard or a runtime check to ensure that `item[key]` is indeed a string before casting it.